compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -1188,4 +1236,46 @@ private boolean isWithinParenthesis(Token openParen, Token closeParen) {
         return !isParenthesisMissing(openParen, closeParen) &&
                 isPosWithinOpenCloseLineRanges(linePosition, openParen.lineRange(), closeParen.lineRange());
     }
+
+    private Optional<TypeSymbol> transformFunctionOrMethod(BLangNode bLangNode, BType variableType,
+                                                           BInvokableSymbol bLangOriginalLangLib,
+                                                           SeparatedNodeList<FunctionArgumentNode> arguments,
+                                                           Token openParenToken, Token closeParenToken,
+                                                           boolean hasFirstArg) {
+        if (bLangOriginalLangLib == null || bLangOriginalLangLib.params.size() == 0) {
+            return getExpectedTypeFromFunction(bLangNode, openParenToken, closeParenToken);
+        }
+
+        BVarSymbol firstParam = bLangOriginalLangLib.params.get(0);
```

Comment:
Are we certain that the the first param will aways be of type `function` with type params. If so shall we specify that using a comment.